airflow/hooks/http_hook.py

```diff
@@ -108,13 +108,13 @@ def run(self, endpoint, data=None, headers=None, extra_options=None):
             url = (self.base_url or '') + (endpoint or '')
 
         req = None
-        if self.method == 'GET':
+        if self.method.upper() == 'GET':
```

Comment:
I would move this to https://github.com/apache/airflow/pull/5173/files#diff-80514189dfbbac3803594380c3a714f1R47 

```
    def __init__(
        self,
        method='POST',
        http_conn_id='http_default'
    ):
        self.http_conn_id = http_conn_id
        self.method = method.upper()
        self.base_url = None
        self._retry_obj = None
```